Microsoft.Azure.Cosmos/src/Telemetry/ClientTelemetryPayloadWriter.cs

```diff
@@ -37,6 +39,11 @@ internal static class ClientTelemetryPayloadWriter
             {
                 foreach (KeyValuePair<OperationInfo, (LongConcurrentHistogram latency, LongConcurrentHistogram requestcharge)> entry in operationInfoSnapshot)
                 {
+                    if (cancellationToken.IsCancellationRequested)
```

Comment:
Replace all the `throw new TimeoutException` with

```suggestion
                    cancellationToken.ThrowIfCancellationRequested();
```